### ISSUE TITLE
Metricbeat errors due to missing `admin/aliases/get` permission for `remote_monitoring_user` user

### ISSUE THREAD

**Issue by mark-vieira** [elastic/elasticsearch#63203]
The default metricbeat configuration using the built-in `remote_monitoring_user` user fails due to a missing permission.

The workaround is to create a separate user with the appropriate permissions but the expectation should be that using the built-in users should "just work". My understand would be that either a) metricbeat is doing something it shouldn't or b) the built-in user should have this permission added.

```
Connection marked as failed because the onConnect callback failed: failed to check for alias 'metricbeat-7.9.1': (status=403) {"error":{"root_cause":[{"type":"security_exception","reason":"action [indices:admin/aliases/get] is unauthorized for user [remote_monitoring_user]"}],"type":"security_exception","reason":"action [indices:admin/aliases/get] is unauthorized for user [remote_monitoring_user]"},"status":403}: 403 Forbidden: {"error":{"root_cause":[{"type":"security_exception","reason":"action [indices:admin/aliases/get] is unauthorized for user [remote_monitoring_user]"}],"type":"security_exception","reason":"action [indices:admin/aliases/get] is unauthorized for user [remote_monitoring_user]"},"status":403}
```

**Comment by tvernum** [comment]
@ycombinator Can you confirm what we need to support here?
 
From previous conversation, it seems to be:
- `GET /_alias/metricbeat-*`
- `HEAD /metricbeat-*`

Is that all?
We can manage the PR once we're confident about what needs to be fixed.

**Comment by ycombinator** [comment]
tl;dr: Yes, those are the only two API calls we are currently missing support for. Further, support for them should be added to the `remote_monitoring_agent` role (**not** the `remote_monitoring_collector` role). For details on how I arrived at this conclusion, see below.

---

I started with a fresh install of Elasticsearch and Metricbeat. Using Wireshark I looked at the HTTP API calls Metricbeat makes to Elasticsearch right after start up (which is when it performs any index management related setup actions). Here's what that looks like:

<img width="1205" alt="Screen Shot 2020-10-08 at 11 21 20 AM" src="https://user-images.githubusercontent.com/51061/95498400-9d8d7e80-0958-11eb-8c89-a15cc138c0ce.png">

Looking at the definition of the `remote_monitoring_user` built-in user, I see that it is assigned the `remote_monitoring_agent` and `remote_monitoring_collector` roles:

https://github.com/elastic/elasticsearch/blob/5197043c41d7a7af3b4df6f11a4f57a5b8f1f88c/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/user/RemoteMonitoringUser.java#L13-L22

https://github.com/elastic/elasticsearch/blob/5197043c41d7a7af3b4df6f11a4f57a5b8f1f88c/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/user/UsernamesField.java#L29-L31

Looking at the definition of these roles...

https://github.com/elastic/elasticsearch/blob/5197043c41d7a7af3b4df6f11a4f57a5b8f1f88c/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java#L70-L99

... I see that they have the following privileges already:

* Cluster-level: 
   * `manage_index_templates`, which resolves to:
      * `indices:admin/template/*`
      * `indices:admin/index_template/*`,
      * `cluster:admin/component_template/*`
   * `manage_ingest_pipelines`, which resolves to:
      * `cluster:admin/ingest/pipeline/*`
   * `monitor`, which resolves to:
      * `cluster:monitor/*`
   * `cluster:admin/ilm/get`
   * `cluster:admin/ilm/put`
* Index-level:
  * On all (`*`) indices:
    * `monitor`, which resolves to:
       * `indices:monitor/*`
  * On `.monitoring-*`:
    * `all`
  * On `metricbeat-*`:
    * `index`, which resolves to:
       * `indices:data/write/index*`
       * `indices:data/write/bulk*`
       * `indices:data/write/update*`
    * `create_index`, which resolves to:
       * `indices:admin/create`
       * `indices:admin/auto_create`
       * `indices:admin/data_stream/create`
  * On `.kibana`:
     *`read`, which resolves to:
        * `indices:data/read/*`

Comparing the API calls made by Metricbeat to Elasticsearch at start up with the privileges already available to the `remote_monitoring_user` built-in user, it appears we're missing privileges to support the `GET /_alias/metricbeat-*` and  `HEAD /metricbeat-*` API calls. These could be satisfied by the `indices:admin/aliases/get` and `indices:admin/get` index-level privileges or the broader `view_index_metadata` index-level privilege.

As noted earlier, the `remote_monitoring_user` built-in user is assigned the `remote_monitoring_agent` and `remote_monitoring_collector` roles. Of these, the former is intended for use by Metricbeat to index _monitored_ data into `.monitoring-*` and `metricbeat-*` indices. The latter is intended for Metricbeat to _monitor_ Elasticsearch and gather data from it. As such, the missing privileges should be granted to the former role, not the latter one.
